File: solutions/121.best-time-to-buy-and-sell-stock.py
#
# @lc app=leetcode id=121 lang=python
#
# [121] Best Time to Buy and Sell Stock
#

# @lc code=start
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min = prices[0]
        res = 0

        for price in prices[1:]:
            if price - min > res:
                res = price - min
            if price < min:
                min = price
        return res
        
# @lc code=end

File: solutions_java/arrays/121.best-time-to-buy-and-sell-stock.java
package solutions_java.arrays;
/*
 * @lc app=leetcode id=121 lang=java
 *
 * [121] Best Time to Buy and Sell Stock
 */

// @lc code=start
class Solution {
    public int maxProfit(int[] prices) {
        int bestPrice = prices[0];
        int maxProfit = 0;
        for (int price : prices) {
            if (price - bestPrice > maxProfit) {
                maxProfit = price - bestPrice;
            }
            if (price < bestPrice) {
                bestPrice = price;
            }
        }
        return maxProfit;
    }
}
// @lc code=end

